### DISCUSSION TITLE
Programmatically creating `mdxjsEsm` nodes

### DISCUSSION THREAD

**Discussion by stefanprobst** [mdx-js/mdx#1800] [Q&A] [answered]
hi, is there currently an official way to programmatically create `mdxjsEsm` nodes in plugins (something similar to e.g. `hastscript`)? what i am doing now is copying an estree from ast-explorer, but was wondering if there is a less verbose way to do this? thanks!

**Comment by wooorm** [comment, marked as answer]
Hmm, not really that I know of. Searching github finds: https://github.com/search?q=estree+build. I’d appreciate it if anyone makes something for that, we could maintain it in the `syntax-tree` org?

I personally just look at AST explorer indeed ;)
